gen/utils/data/class.go

```diff
@@ -0,0 +1,175 @@
+package data
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"unicode"
+
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+type Class struct {
+	// This is used to prevent the deletion of the class if it is not allowed on APIC.
+	AllowDelete bool
+	// Full name of the class, ex "fvTenant".
+	ClassName string
+	// Capitalized name of the class, ex "FvTenant".
+	ClassNameForFunctions string
+	// Package part of the class, ex "fv".
+	ClassNamePackage string
+	// Name part of the class, ex "Tenant".
+	ClassNameShort string
+	// List of all child classes which are included inside the resource.
+	// When looping over maps in golang the order of the returned elements is random, thus list is used for order consistency.
+	Children []Child
+	// List of all possible parent classes.
+	ContainedBy []string
+	// Documentation specific information for the class.
+	Documentation ClassDocumentation
+	// List of all identifying properties of the class.
+	// These are properties that are part if the relative name (RN) format, ex 'tn-{name}' where name is the identifying property.
+	IdentifiedBy []string
+	// Indicates that the class is migrated from previous version of the provider.
+	// This is used to determine if legacy attributes to be exposed in the resource.
+	// This is used to determine if the plan modification logic.
```

Comment:
```suggestion
	// This is used to determine if the plan modification logic.
```
Is this needed here? Not sure why 2 "this is used"